### ISSUE TITLE
Is MODBUS_REGQ_X_CHECK specified or designed?

### ISSUE THREAD

**Issue by f-zl** [Jacajack/liblightmodbus#31]
In the slave code, when a request is received, the user-defined register callback is called for every registers with arg type `MODBUS_REGQ_X_CHECK` (where X is either R or W), then the register back is called again for every register with arg type `MODBUS_REGQ_X`.

In `modbusParseRequest01020304` one comment says the check is for "Check if all registers can be read". I'm not a modbus expert, but I fail to find information about the first check then read/write behavior in modbus specifications. Is it a specified behavior or a design in implementation?

**Comment by Jacajack** [comment]
That's correct - as far as I know, there's no mention of read/write protected registers in the Modbus specification. However, the process of writing a register looks like this in the [Modbus Application Protocol document](https://modbus.org/docs/Modbus_Application_Protocol_V1_1b.pdf):
<img src=https://user-images.githubusercontent.com/6173262/199326151-df4d581e-518c-4868-ac28-34095c3f6085.png width=500px />

Note the highlighted step - if the register cannot be written for one reason or another, the slave can return `MODBUS_EXCEP_SLAVE_FAILURE`. Register read/write protection in liblightmodbus is based on that. I'd say that  "read/write check" is an implementation specific term, but the Modbus protocol definitely provides a way for the slaves to deny a request due to internal reasons.
